### ISSUE TITLE
parser: select expression doesn't work

### ISSUE THREAD

**Issue by TimSatke** [tomarrell/lbadd#146]
Try to parse `SELECT amount * price AS total_price FROM items`. This will throw a lot of errors, although the production

`SELECT result-column ...`

with `result_column` using the `expr` production, and `expr` using the `expr binary-operator expr` allows such a statement.

**Comment by SUMUKHA-PK** [comment]
Okay two things.

1. `SELECT amount * price AS total_price FROM items` is a malformed statement. The allowed version of the statment is `SELECT amount, * , price AS total_price FROM items` (result column's must be separated by `,`'s)

2. There is a scanner bug which doesnt allow literals with underscores `_`, which needs to be fixed for this statement to be parsed.


Here's the parser run for `SELECT amount,*, price AS total FROM items` as a proof:

```
parser_test.go:9648:   &ast.SQLStmt{
              	... // 23 identical fields
              	RollbackStmt:  nil,
              	SavepointStmt: nil,
            - 	SelectStmt:    nil,
            + 	SelectStmt: &ast.SelectStmt{
            + 		SelectCore: []*ast.SelectCore{
            + 			&{
            + 				Select: s"KeywordSelect(SELECT)",
            + 				ResultColumn: []*ast.ResultColumn{
            + 					&{Expr: &ast.Expr{LiteralValue: s"Literal(amount)"}},
            + 					&{Asterisk: s"BinaryOperator(*)"},
            + 					&{
            + 						Expr:        &ast.Expr{LiteralValue: s"Literal(price)"},
            + 						As:          s"KeywordAs(AS)",
            + 						ColumnAlias: s"Literal(total)",
            + 					},
            + 				},
            + 				From:       s"KeywordFrom(FROM)",
            + 				JoinClause: &ast.JoinClause{TableOrSubquery: &ast.TableOrSubquery{TableName: s"Literal(items)"}},
            + 			},
            + 		},
            + 	},
              	UpdateStmt:        nil,
              	UpdateStmtLimited: nil,
              	VacuumStmt:        nil,
              }
```

**Comment by TimSatke** [comment]
It is not a malformed statement, production explanation is added in the original description.
`amount * price` is a binary expression, which is an expression, which is a result column.

**Comment by TimSatke** [comment]
This was not fixed by #149

**Comment by SUMUKHA-PK** [comment]
Yes, will have a patch soo.
